python/cudf/cudf/_cuda/gpu.pyx

```diff
@@ -0,0 +1,272 @@
+    cudaDevAttrLocalL1CacheSupported = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrLocalL1CacheSupported
+    cudaDevAttrMaxSharedMemoryPerMultiprocessor = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrMaxSharedMemoryPerMultiprocessor
+    cudaDevAttrMaxRegistersPerMultiprocessor = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrMaxRegistersPerMultiprocessor
+    cudaDevAttrManagedMemory = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrManagedMemory
+    cudaDevAttrIsMultiGpuBoard = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrIsMultiGpuBoard
+    cudaDevAttrMultiGpuBoardGroupID = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrMultiGpuBoardGroupID
+    cudaDevAttrHostNativeAtomicSupported = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrHostNativeAtomicSupported
+    cudaDevAttrSingleToDoublePrecisionPerfRatio = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrSingleToDoublePrecisionPerfRatio
+    cudaDevAttrPageableMemoryAccess = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrPageableMemoryAccess
+    cudaDevAttrConcurrentManagedAccess = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrConcurrentManagedAccess
+    cudaDevAttrComputePreemptionSupported = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrComputePreemptionSupported
+    cudaDevAttrCanUseHostPointerForRegisteredMem = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrCanUseHostPointerForRegisteredMem
+    cudaDevAttrReserved92 = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrReserved92
+    cudaDevAttrReserved93 = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrReserved93
+    cudaDevAttrReserved94 = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrReserved94
+    cudaDevAttrCooperativeLaunch = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrCooperativeLaunch
+    cudaDevAttrCooperativeMultiDeviceLaunch = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrCooperativeMultiDeviceLaunch
+    cudaDevAttrMaxSharedMemoryPerBlockOptin = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrMaxSharedMemoryPerBlockOptin
+    cudaDevAttrCanFlushRemoteWrites = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrCanFlushRemoteWrites
+    cudaDevAttrHostRegisterSupported = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrHostRegisterSupported
+    cudaDevAttrPageableMemoryAccessUsesHostPageTables = \
+        <c_attr> \
+        cudaDeviceAttr.cudaDevAttrPageableMemoryAccessUsesHostPageTables
+    cudaDevAttrDirectManagedMemAccessFromHost = \
+        <c_attr> cudaDeviceAttr.cudaDevAttrDirectManagedMemAccessFromHost
+
+
+def driverGetVersion():
+    """
+    Returns in the latest version of CUDA supported by the driver.
+    The version is returned as (1000 major + 10 minor). For example,
+    CUDA 9.2 would be represented by 9020. If no driver is installed,
+    then 0 is returned as the driver version.
+
+    This function returns -1 if driver version is NULL.
+    """
+    cdef int version
+    status = cudaDriverGetVersion(&version)
+    return -1 if status != 0 else version
```

Comment:
We should throw an exception here if the status is non zero instead of returning an error code from here